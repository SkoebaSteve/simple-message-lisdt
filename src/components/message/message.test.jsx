/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import sinon from 'sinon'
import toJson from 'enzyme-to-json'
import Message from './'
import Button from '../button'

const message = {
  id: '1',
  type: 'important',
  date: '2018-02-16',
  content: 'this is a very important message',
}

it('Has the default values.', () => {
  const wrapper = shallow(<Message {...message} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('Has the liked but ton active.', () => {
  const wrapper = shallow(<Message {...message} liked />)
  expect(wrapper.find(Button).at(0).hasClass('active')).toEqual(true)
  expect(wrapper.find(Button).at(1).hasClass('active')).toEqual(false)
})

it('Has the disliked button active.', () => {
  const wrapper = shallow(<Message {...message} liked={false} />)
  expect(wrapper.find(Button).at(0).hasClass('active')).toEqual(false)
  expect(wrapper.find(Button).at(1).hasClass('active')).toEqual(true)
})

it('throws a click event', () => {
  const onButtonClick = sinon.spy()
  const wrapper = shallow(<Message {...message} changeLiked={onButtonClick} />)
  wrapper.find(Button).at(0).simulate('click')
  expect(onButtonClick).toHaveProperty('callCount', 1)
  wrapper.find(Button).at(1).simulate('click')
  expect(onButtonClick).toHaveProperty('callCount', 2)
})
