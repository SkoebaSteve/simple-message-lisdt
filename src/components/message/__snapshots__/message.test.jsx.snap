// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Has the default values. 1`] = `
<div>
  <div>
    important
     | 
    <t
      ago={false}
      calendar={false}
      decimal={false}
      element="time"
      filter={[Function]}
      format="MMMM DD"
      fromNow={false}
      interval={60000}
      onChange={[Function]}
      toNow={false}
      unit={null}
      unix={false}
      utc={false}
    >
      2018-02-16
    </t>
  </div>
  <p>
    this is a very important message
  </p>
  <Button
    className={false}
    label="like this"
    onClick={[Function]}
  />
  <Button
    className={false}
    label="dislike this"
    onClick={[Function]}
  />
</div>
`;
